### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by AntonioHincapie**
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond to JSON as well.

### CODE REVIEW

File: app/controllers/api/post_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Api::PostsController < ApplicationController
+  def index
+    user = User.find(params[:user_id])
+    posts = user.posts.order(Arel.sql('created_at DESC'))
+    render json: { posts: }, status: :ok
```
Comment (FrederickMih) [inline app/controllers/api/post_controller.rb:5]: - [x] Request for all posts in a JSON format using RESTful API are not working. That goes for comments. Please kindly fix these issues. I didn't get any response to this GET request: http://localhost:3000/api/users/:user_id/posts or http://localhost:3000/api/users/:user_id/posts/:id/comments

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AntonioHincapie , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- PR has a title and summary :+1: 
- GitHub workflow has been followed :+1: 
- No Linter error :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AntonioHincapie** [comment]
Hi, I solve the issue.

As you can see it is working now.

image

Thanks for your help.

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @AntonioHincapie , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

### Highlights:
 ✔️ Project is setup correctly
 ✔️ All linters are passing
✔️ Correct gitflow
 ✔️ Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

image

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@shoirata] in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**